tests/main/security-devpts/pts.exp

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/expect
+
+set timeout 2
+
+send "ls /dev/pts"
+expect "\[0-9\]*ptmx"
+
+# Launch app and checks contents of /dev/pts, should have only ptmx
+spawn sudo -i -u test /bin/sh -c /snap/bin/test-snapd-tools.sh
+expect "bash-4.3$"
+send "ls /dev/pts"
+expect "^ptmx"
+
+# From within confined app, open a pty
+send "python3"
+expect ">>>"
+send "import os, pty, sys"
+send "os.listdir('/dev/pts')"
+expect "\['ptmx'\]"
+send "pty.openpty()"
+send "os.listdir('/dev/pts')"
+expect "\['0', 'ptmx'\]"
+send "sys.exit(0)"
+expect "bash-4.3$"
+
+# From with confined app, verify pty was closed (assumes that the last program
+# closes it properly, which the above does)
+send "ls /dev/pts"
+expect "^ptmx"
```

Comment:
sorry, I'm quite new to expect and need to learn a lot yet. I've made a couple of changes to your script (use test-snapd-tools instead of hello-world, launching the second spawn as the test user instead of the, default, root user and removing the last exit call), it passes executing locally, please take a look.